src/client/Client.cc

```diff
@@ -11640,7 +11644,6 @@ int Client::_setxattr(Inode *in, const char *name, const void *value,
 	if (ret < 0)
 	  return ret;
 	if (ret == 0) {
-	  value = NULL;
```

Comment:
So the change to `_do_setxattr` is fine I think but this code is implicitly relying on the xattr being removed, not set to the empty string. This change in behavior appears to break many tests, including the pjd suite:

http://pulpito.ceph.com/pdonnell-2020-06-24_22:47:27-fs-wip-pdonnell-testing-20200624.212840-distro-basic-smithi/5176636/

and many others. I had made a ticket here: https://tracker.ceph.com/issues/46213

Now I believe the thing to do here is to have this method instead make a `MetaRequest(CEPH_MDS_OP_RMXATTR)` request. That will probably involve calling`::_removexattr`.

Please confirm this works with the pjd tests before asking for review.

Here's a run of the current master for comparison: http://pulpito.ceph.com/pdonnell-2020-06-25_19:18:41-fs-master-distro-basic-smithi/

Reply:
Thanks. Let me test it out.